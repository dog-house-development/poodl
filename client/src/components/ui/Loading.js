import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    kind: PropTypes.oneOf(['primary', 'secondary']),
    content: PropTypes.string
};

const defaultProps = {
    size: 'medium',
    kind: 'primary',
    content: 'Loading...'
};

class Loading extends Component {
    render() {
        return (
            <div className="loading-container">
                <div className={`loading ${this.props.size} ${this.props.kind}`}>
                    <div />
                </div>
                <p>{this.props.content}</p>
            </div>
        );
    }
}

Loading.propTypes = propTypes;
Loading.defaultProps = defaultProps;
export default Loading;
