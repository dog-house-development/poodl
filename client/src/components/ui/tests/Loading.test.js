import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';

import Loading from '../Loading';

configure({ adapter: new Adapter() });

describe('Loading tests', () => {
    let wrapper, instance;
    const setInstanceAndWrapper = (_props = {}) => {
        const props = _.assign(
            {},
            {
                size: 'medium',
                kind: 'primary'
            },
            _props
        );
        wrapper = shallow(<Loading {..._.assign({}, props)} />);
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('render', () => {
        it('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
