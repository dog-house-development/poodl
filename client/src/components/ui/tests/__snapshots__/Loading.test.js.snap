// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading tests render should render correctly 1`] = `
<div
  className="loading-container"
>
  <div
    className="loading medium primary"
  >
    <div />
  </div>
  <p>
    Loading...
  </p>
</div>
`;
